### PR TITLE
Flask start project

### PR DESCRIPTION
**Pull request by denysgerasymuk799**
Flask

### CODE REVIEW

File: web-server/app.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
+from flask import Flask, render_template, request
+from os import path
```
Comment (sofiia-tesliuk) [inline web-server/app.py]: remove unnecessary lines (you don't use path form os).

File: web-server/app.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+app = Flask(__name__)
+tasks = ["eat", "sleep", "have fun", "learn", "grow", "code", "compete", "dream", "travel", "help"]
```
Comment (sofiia-tesliuk) [inline web-server/app.py]: remove unnecessary line (you don't use this list).

File: web-server/app.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+
+@app.route('/')
+def render_main_page():
+    return render_template('main.html')
```
Comment (sofiia-tesliuk) [inline web-server/app.py]: Please, use for strings double quotes. Replace ' to " in every string.

File: web-server/static/style.css [hunk 4]
```diff
@@ -0,0 +1,19 @@
+body {
+    background-color: aliceblue;
+    font-family: "Helvetica Neue", Helvetica, Arial, sans-serif;
```
Comment (sofiia-tesliuk) [inline web-server/static/style.css]: Please remove here font-family. You have applied another font-family below.

File: web-server/templates/result.html [hunk 5]
```diff
@@ -0,0 +1,15 @@
+<!DOCTYPE html>
```
Comment (sofiia-tesliuk) [inline web-server/templates/result.html]: delete this unused file.

**Review by sofiia-tesliuk — APPROVED** [review]

File: web_server/registration.py [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+
+@registration_page.route("/", methods=["GET", "POST"])
+def registration():
+    if request.method == "POST":
+        if request.form.get("name"):
+            if request.form.get("name") not in users:
+                print('user is ok')
+
+                if request.form.get("email"):
```
Comment (sofiia-tesliuk) [inline web_server/registration.py]: Too much nested "if" conditions. Try to make it less complex.

For example:
``` 

if not request.form.get("name"):
    return render_template('registration.html', error="enter name")
if not request.form.get("phone_number"):
    return render_template('registration.html', error="enter phone number")

return render_template('registration.html')
```

File: web_server/login.py [hunk 7]
```diff
@@ -15,8 +15,13 @@ def login():
                 return redirect(url_for('login.hello'))
             else:
                 print('password not ok')
```
Comment (sofiia-tesliuk) [inline web_server/login.py]: delete unnecessary debug prints

File: web_server/registration.py [hunk 8]
```diff
@@ -9,37 +9,21 @@
...
+
+        parameters = ["name", "email", "phone number", "password"]
+        for parameter in parameters:
+            if not request.form.get(parameter):
+                error = 'enter {}'.format(parameter)
+                return render_template('registration.html', error=error)
+
+            if parameter == "name":
+                users[request.form.get("name")] = []
+            elif parameter != "name":
```
Comment (sofiia-tesliuk) [inline web_server/registration.py:21]: Just `else:` is enough here.

File: web_server/registration.py [hunk 9]
```diff
@@ -9,37 +9,21 @@
...
-                    print('enter email')
-                    error = 'enter email'
-                    return render_template('registration.html', error=error)
-
-        else:
-            print('enter username')
-            error = 'enter username'
-            return render_template('registration.html', error=error)
+
+        parameters = ["name", "email", "phone number", "password"]
```
Comment (sofiia-tesliuk) [inline web_server/registration.py]: Take parameters variable out of function.
Do not initialize it each time, when function is called.

File: web_server/registration.py [hunk 10]
```diff
@@ -9,37 +9,21 @@
...
+        for parameter in parameters:
+            if not request.form.get(parameter):
+                error = 'enter {}'.format(parameter)
+                return render_template('registration.html', error=error)
+
+            if parameter == "name":
+                users[request.form.get("name")] = []
+            elif parameter != "name":
+                users[request.form.get("name")].append(request.form.get(parameter))
```
Comment (sofiia-tesliuk) [inline web_server/registration.py]: Save information related to users also as dictionary, not as list.

File: web_server/login.py [hunk 11]
```diff
@@ -9,16 +9,12 @@
 def login():
     if request.method == "POST":
         if request.form.get("name") in users:
-            print('user is ok')
             if request.form.get("password") == users[request.form.get("name")]:
```
Comment (sofiia-tesliuk) [inline web_server/login.py]: Check here if password correspond to user as you save information related to users.
For example, if you save info as dictionary, you need to access `password`:  `users[request.form.get("name")]["password"]`.

**Review by sofiia-tesliuk — APPROVED** [review]

File: web_server/static/style_for_todo_list.css [hunk 12]
```diff
@@ -0,0 +1,27 @@
...
+
+main.new_task_box > form > * {
+    height: 1.5rem;
+}
+
+main.new_task_box > form > button {
+    background-color: darkturquoise;
+    color: whitesmoke;
+    font-weight: bold;
+}
```
Comment (LAnastasiia) [inline web_server/static/style_for_todo_list.css]: There's no need to use this stylesheet in `hello.html`. You can just delete it. We can keep the others for now, but this one was specific for the todo-list example from the workshop.

File: web_server/app.py [hunk 13]
```diff
@@ -0,0 +1,19 @@
...
+
+
+@app.route("/")
+def render_main_page():
+    return redirect(url_for('login'))
+
+
+if __name__ == '__main__':
+    app.run(debug=True)
+
```
Comment (LAnastasiia) [inline web_server/app.py:19]: Nice work! I really like how you broke it into several modules.

File: web_server/login.py [hunk 14]
```diff
@@ -0,0 +1,26 @@
+from flask import Blueprint, render_template, request, redirect, url_for
+
+login_page = Blueprint('login', __name__, template_folder='templates')
+
+users = {'sdv': {'email': 'sdv', 'phone number': 'sdv', 'password': 'sdv'}}
```
Comment (LAnastasiia) [inline web_server/login.py:5]: Do we want to push the test-data? 
Maybe just leave it an empty dictionary (and add a comment about its structure if you'd like).

**Review by LAnastasiia — APPROVED** [review]
great work!

**Pull request merged**